docs/control-plane-architecture.adoc

```diff
@@ -0,0 +1,252 @@
+* On the two Scrimlets, a "switch zone" provides additional functionality related to the switch:
+** **Dendrite** provides APIs for configuring the switch itself (e.g., populating various tables used for packet forwarding, NAT, etc.).  
+** **Management Gateway Service (MGS)** provides APIs for communicating with all the rack's service processors (including those on the sleds, Sidecars, and PSCs).  See <<rfd210>> for details.
+** **Wicket** and its associated service **wicketd** provide a text user interface (TUI) that's accessible over the rack's technician ports.  Wicket is used for initial system setup (before networking has been configured) and for support.
+** **Boundary NTP** provides NTP service for all sleds in the rack based on upstream NTP servers provided by the customer.
+
+.Components deployed alongside specific hardware
+[cols="1h,2,4",stripes="none",options="header"]
+|===
+| Component
+| How it's deployed
+| Availability/scalability
+
+| Sled agent
+| One per sled, tied to that specific sled
+| N/A
+
+| Internal DNS
+| One zone per non-Scrimlet sled
+| N/A
+
+| Boundary NTP
+| One zone per Scrimlet.  Both instances within a rack are fungible.
+| There are two.  Short-term failure (order of hours or even days) is unlikely to affect anything since sled clocks do not drift that quickly.
+
+| Dendrite
+| Part of the switch zone (one per Scrimlet), tied to that specific switch
+| Unavailability of either instance results in loss of ability to configure and monitor the corresponding switch.
+
+| Management Gateway
+| Part of the switch zone (one per Scrimlet)  Both instances within one rack are fungible.
+| Only one of the two instances are generally required to maintain service.
+
+| Wicket
+| Part of the switch zone (one per Scrimlet).  Both instances within one rack are fungible.
+| Wickets operate independently.  Failure of one means unavailability of the TUI over that technician port.
+
+|===
+
+=== Higher-level components
+
+Most other components:
+
+* are deployed in illumos zones
+* don't care where they run and can even be deployed multiple times on the same sled
+* can be deployed multiple times for availability, horizontal scalability, or both
+
+They are:
+
+* **Nexus** provides primary control for the whole control plane.  Nexus hosts all user-facing APIs (both operator and customer), the web console, and internal APIs for other control plane components to report inventory, generate alerts, and so on.  Nexus is also responsible for background control plane activity, including utilization management, server failure detection and recovery, and the like.  Persistent state is stored elsewhere (in CockroachDB), which allows us to be scaled separately.
```

Comment:
Could be more clear about what "us" means.

Reply:
Heh, yeah, that seems like a typo for "Nexus".